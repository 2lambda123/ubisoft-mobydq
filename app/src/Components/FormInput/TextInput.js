import React from 'react';
import PropTypes from 'prop-types';
import TextField from '@material-ui/core/TextField';
import FormControlLabel from '@material-ui/core/FormControlLabel';
import InputFeedback from './InputFeedback';

const TextInput = ({
  id,
  label,
  helperText,
  error,
  touched,
  errors,
  value,
  onChange,
  className,
  ...props
}) => {
  return (
    <FormControlLabel
      control={
        <React.Fragment>
          <TextField
            id={id}
            label={label}
            helperText={helperText}
            onChange={onChange}
            value={value}
            // errorText={touched && errors}
            {...props}
          />
          <InputFeedback error={error} />
        </React.Fragment>
      }
      label={""}
    />
  );
};

export default TextInput;


TextInput.propTypes = {
  id: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  helperText: PropTypes.string.isRequired,
};