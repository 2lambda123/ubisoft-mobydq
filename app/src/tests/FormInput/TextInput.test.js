import React from 'react'
import * as Adapter from 'enzyme-adapter-react-16';
import { shallow, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import TextInput from '../../Components/FormInput/TextInput';

describe('TextInput unit test', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <TextInput
        id="Id"
        label="Label"
        helperText="Helper Text"
        placeholder="Enter indicator name"
        touched={jest.fn()}
        error=""
        value="value"
        onChange={jest.fn()}
        onBlur={jest.fn()}
      />
    );
  });

  it('renders', () => {
    expect(wrapper.length).toEqual(1);
  });

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe('TextInput functional test', () => {
  let wrapper = mount(
      <TextInput
        id="Id"
        label="Label"
        helperText="Helper Text"
        placeholder="Enter indicator name"
        touched={jest.fn()}
        error=""
        value="mock text"
        onChange={jest.fn()}
        onBlur={jest.fn()}
      />
    );
  it ('renders correct contents', () => {
    expect(wrapper.prop('value')).toEqual('mock text')
  });
  it ('renders correct contents', () => {
    wrapper.setProps({value: 'new mock content'})
    expect(wrapper.prop('value')).toEqual('new mock content')
  });
  it ('renders correct numeric contents', () => {
    wrapper.setProps({numeric: true, value: '123321'});
    expect(wrapper.prop('value')).toEqual('123321');
  });
});
