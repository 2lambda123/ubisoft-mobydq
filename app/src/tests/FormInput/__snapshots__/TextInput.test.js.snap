// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput unit test matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TextInput
    error=""
    helperText="Helper Text"
    id="Id"
    label="Label"
    onBlur={[MockFunction]}
    onChange={[MockFunction]}
    placeholder="Enter indicator name"
    touched="true"
    value="value"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <WithStyles(FormControlLabel)
          control={
            <React.Fragment>
              <TextField
                fullWidth={true}
                helperText="Helper Text"
                id="Id"
                label="Label"
                margin="dense"
                onBlur={[MockFunction]}
                onChange={[MockFunction]}
                placeholder="Enter indicator name"
                required={false}
                select={false}
                type="text"
                value="value"
                variant="outlined"
              />
            </React.Fragment>
          }
          label=""
        />,
        <WithStyles(FormHelperText)>
          
        </WithStyles(FormHelperText)>,
      ],
      "error": false,
      "style": Object {
        "width": "500px",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "control": <React.Fragment>
            <TextField
              fullWidth={true}
              helperText="Helper Text"
              id="Id"
              label="Label"
              margin="dense"
              onBlur={[MockFunction]}
              onChange={[MockFunction]}
              placeholder="Enter indicator name"
              required={false}
              select={false}
              type="text"
              value="value"
              variant="outlined"
            />
          </React.Fragment>,
          "label": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "",
        },
        "ref": null,
        "rendered": "",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <WithStyles(FormControlLabel)
            control={
              <React.Fragment>
                <TextField
                  fullWidth={true}
                  helperText="Helper Text"
                  id="Id"
                  label="Label"
                  margin="dense"
                  onBlur={[MockFunction]}
                  onChange={[MockFunction]}
                  placeholder="Enter indicator name"
                  required={false}
                  select={false}
                  type="text"
                  value="value"
                  variant="outlined"
                />
              </React.Fragment>
            }
            label=""
          />,
          <WithStyles(FormHelperText)>
            
          </WithStyles(FormHelperText)>,
        ],
        "error": false,
        "style": Object {
          "width": "500px",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "control": <React.Fragment>
              <TextField
                fullWidth={true}
                helperText="Helper Text"
                id="Id"
                label="Label"
                margin="dense"
                onBlur={[MockFunction]}
                onChange={[MockFunction]}
                placeholder="Enter indicator name"
                required={false}
                select={false}
                type="text"
                value="value"
                variant="outlined"
              />
            </React.Fragment>,
            "label": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "",
          },
          "ref": null,
          "rendered": "",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
